src/actions/action.surrounder.ts

```diff
@@ -0,0 +1,72 @@
+import { Context } from "telegraf";
+import { Message, User } from "telegraf/typings/core/types/typegram";
+import { AxiosError } from "axios";
+
+export default abstract class Action {
+  id: string;
+  message: Message.CommonMessage;
+  inline_message_id: string;
+  from: User;
+  item_name: string;
+  context: Context;
+
+  get ctx() {
+    return this.context;
+  }
+
+  set ctx (ctx: Context) {
+    this.context = ctx;
+    this.id = (ctx.callbackQuery as any).data.split(':')[1];
+    this.message = ctx.callbackQuery.message;
+    this.inline_message_id = ctx.callbackQuery.inline_message_id;
+    this.from = ctx.callbackQuery.from;
+  }
```

Comment:
I think that this class should not store `ctx` value and its internal state as its own state. Instead, it should receive `ctx` as a parameter in the `execute` method.

There is no problem with accidental `ctx` pollution or using wrong `ctx` instance.